test/src/utils/vectors/vector_basic.test.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Author: Pablo Morandé & Flori
+ * Organisation: HYPED
+ * Date: 13/11/2020
+ * Description: Testing file for vector.cpp
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *    except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *    either express or implied. See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <math.h>
+#include <cstdlib>
+#include "gtest/gtest.h"
+#include "utils/math/vector.hpp"
+#include "vector_operations_with_vectors.test.cpp"
+
+
+
+namespace hyped
+{
+namespace utils
+{
+namespace math
+{
+/**
+ * @brief Struct used to set up all the variables used in the tests for basic operationsof a vector
+ * with constants.
+ * This tests will check if that basic operations with constants are performed the way  they should.
+ * However, they do not test properties.
+ */
+struct OperationsByConstant : public::testing::Test
+{
+  const int dimension = 3;
+  Vector<int, 3> vector_one;
+  Vector<int, 3> vector_two;
+  Vector<int, 3> vector_three;
+  Vector<int, 3> vector_result_one;
+  Vector<int, 3> vector_result_two;
+  int kValue = rand()%1000;
+  void SetUp()
+  {
+    vector_one = Vector<int, 3>(createRandomArray());
+    vector_two = Vector<int, 3>(createRandomArray());
+    vector_three = Vector<int, 3>(createRandomArray());
+    vector_result_one = Vector<int, 3>();
+    vector_result_two = Vector<int, 3>();
+  }
+};
+/**
+ * @brief Test used to determine if the current implementation of the class vector allows adding in
+ * a vector with a constant in normal way (adding the constant to each element),
+ * this tests the operator (+).
+ * The test will be performed using one random 3D vector and a random constant.
+ */
+TEST_F(OperationsByConstant, handlesAutoAdditionWithConstant)
```

Comment:
It would be more readable when the tests are running if we mentioned this was VectorOperations rather than just OperationsByConstant.